### DISCUSSION TITLE
Three concentration components diffusion model

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#17008] [Q&A General] [answered]
Hi all,

I am modeling three concentration components diffusion (please see attached).
It would be appreciated if you could provide me with examples (suggestions).

Sincerely,
Thomas
[Three_components_diffusion_model.pdf](https://github.com/idaholab/moose/files/5984795/Three_components_diffusion_model.pdf)

**Comment by WilkAndy** [comment, marked as answer]
Naively, this is just 3 decoupled diffusion equations, so you'd have three Variables, and three `Diffusion` Kernels and three `TimeDerivative` Kernels.

More interestingly, do the diffusion coefficients (D_{1-gas}, etc) depend on the concentrations in some way?  Are they scalars or tensors?  How about the source terms (g_{1}, etc) - are they constants, or do they depend on concentration?

**Reply by styyokuda** [reply]
Thank you, WilkAndy.
Yes, the diffusion coefficients and sources are actually a function of gas, c1, c2, c3, temperature, and time.
The diffusion coefficients are scalars.
But I would like to start the diffusion coefficients and sources as a function of gas and temperature only.
It would be appreciated if you could tell me where I look for the examples.
Sincerely,
Thomas

**Reply by aeslaughter** [reply]
I don't think we have an example of this, but it shouldn't be hard to solve. I would begin by setting up the 3 diffusion equations as @WilkAndy suggested. Then switch over to using the Material system to compute the correct coefficients for each equation: https://mooseframework.inl.gov/syntax/Materials/index.html
